torch/__init__.py

```diff
@@ -1752,6 +1752,90 @@ def registerOp(cls, op_key, full_schema, op_impl, dispatch_key):
             return cls.ops_table[(op_key, dispatch_key)]
 
 
+################################################################################
+# Exposing torch.export() and related support classes
+################################################################################
+
+# TODO(ycao): Move ExportedProgram, Constraint, dynamic_dim etc to torch.compiler namespace
+
+def export(
+    f: Callable,
+    args: Tuple[Any],
+    kwargs: Optional[Dict[str, Any]] = None,
+    *,
+    constraints: Optional[List["torch._dynamo.eval_frame.Constraint"]] = None,
+) -> "torch._export.exported_program.ExportedProgram":
+    """
+    torch.export() is a one-shot process for capturing a computation graph from
+    a PyTorch program Ahead-of-Time (AOT).
+
+    This function traces either an nn.Module's forward function or a callable
+    containing PyTorch operations and produces an ExportedProgram. The
+    ExportedProgram includes PyTorch operations that perform computations
+    equivalent to those in the given nn.Module or callable.
+
+    In specific terms, torch.export() traces a function `f` by executing it
+    with the provided `args` and `kwargs`. It records the PyTorch operations
+    invoked during execution to produce the ExportedProgram.
+
+    Acceptable types of inputs for `args` and `kwargs` include:
+        - PyTree leaf types (such as torch.Tensor, int, float, bool)
```

Comment:
So.... pytree is technically not public API. So it's best to either explicitly spell out exactly what the leaf/flattened types are, or maybe @zou3519 knows if we have some doc saying exactly what is allowed.

I still have not completely fixed the bad error reporting when you pass in a non-pytree'able entity to torch.export, so it is probably worth having a warning about this footgun until we actually manage to fix this.

Reply:
So we should remove all mentions of PyTree and list all specific data structures and types supported?

Reply:
Well, we probably should make pytree public, but if you don't want to yak shave having good docs for pytree, you'll need to recap it here.

Reply:
We don't have a good doc on pytrees, should just recap it here. It's probably easiest to explain it as "nested list/tuple/dict of leaf types (Tensor, int, float, bool)", (list/dict/tuple are the most used, but there are more types somewhere)